packages/analytics/src/utils/getters.ts

```diff
@@ -126,3 +126,32 @@ export const getCookie = (name: string): string | void => {
     return parts.pop()?.split(';').shift();
   }
 };
+
+/**
+ * Obtain checkout order generic fields from analytics properties.
+ *
+ * @param dataProperties - The event's properties data.
+ * @param orderIdFieldName - The name of orderId field. Default as `checkoutOrderId`.
+ * @returns The checkout generic order data result.
+ */
+export const getCheckoutOrderGenericProperties = (
+  dataProperties: EventProperties,
+  orderIdFieldName = 'checkoutOrderId',
+) => {
+  // Valid order code should be an alphanumeric value.
+  // Should be used 'orderId' values like e.g.: 5H5QYB
+  // and for 'checkoutOrderId' values like e.g.:123123123
+  const validOrderCode = isNaN(Number(dataProperties?.orderId));
+  const orderCode = validOrderCode ? dataProperties?.orderId : undefined;
+  const orderId = validOrderCode
+    ? dataProperties?.[orderIdFieldName]
+    : Number(dataProperties?.orderId);
+
+  return {
+    validOrderCode,
+    data: {
```

Comment:
I think it is unnecessary to wrap those 2 fields in a `data` field. You can send then directly on the parent object.
Also you can easily derive the value of `validOrderCode` by checking if the returned `orderCode` field is not undefined, like this, so I would remove it as it is redundant:

```js

const orderIdentificationProperties = getCheckoutOrderIdentificationProperties(data);

const hasValidOrderCode = orderIdentificationProperties.orderCode !== undefined;

if(!hasValidOrderCode) {
....
}


```